lookout/style/format/benchmarks/evaluate_smoke.py

```diff
@@ -1,48 +1,190 @@
 """Module for Smoke dataset evaluation."""
 import csv
 from difflib import SequenceMatcher
+from itertools import chain
 import json
 import logging
+import os
 from pathlib import Path
 import tempfile
-from typing import Any, List, Mapping
+import time
+from typing import Any, List, Mapping, Sequence, Tuple, Union
+
+import pandas
+from tqdm import tqdm
 
 from lookout.core.analyzer import ReferencePointer
 from lookout.core.api.service_analyzer_pb2 import Comment
 from lookout.core.api.service_data_pb2_grpc import DataStub
 from lookout.core.data_requests import with_changed_uasts_and_contents
-from lookout.core.lib import files_by_language, filter_files, find_new_lines
+from lookout.core.lib import files_by_language, filter_files, find_deleted_lines, find_new_lines
 from lookout.core.tests import server
 from lookout.style.format.analyzer import FormatAnalyzer
 from lookout.style.format.code_generator import CodeGenerator
 from lookout.style.format.feature_extractor import FeatureExtractor
 from lookout.style.format.model import FormatModel
 from lookout.style.format.tests.test_analyzer_integration import TestAnalyzer
 
+log = logging.getLogger("report_summary")
 
-def _loss(head_lines, correct_lines, predicted_lines):
-    def edit_distance(a, b):
-        err = 0
-        matcher = SequenceMatcher(a=a, b=b)
-        matcher.ratio()
-        actions = frozenset(("delete", "insert", "replace"))
-        for action, _, _, _, _ in matcher.get_opcodes():
-            if action == "equal":
-                continue
-            if action in actions:
-                err += 1
-        return err
+EMPTY = "␣"
+
+
+def align(seq1: Sequence, seq2: Sequence, seq3: Sequence = None
+          ) -> Union[Tuple[Sequence, Sequence],
+                     Tuple[Sequence, Sequence, Sequence]]:
+    """
+    Align two sequences using levenshtein distance.
```

Comment:
The docstring says "two" and we accept three. You need to add more details to the summary.

Reply:
Makes sense to rename to `align2` and also rename `seq3` to smth else since it has a special logic and input constraints (e.g. same length as seq2).

Reply:
Please check if it is more clear now.